eclair-core/src/test/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisherSpec.scala

```diff
@@ -358,16 +409,196 @@ class ReplaceableTxPublisherSpec extends TestKitBaseClass with AnyFunSuiteLike w
+    }
+  }
+
+  test("commit tx not confirming, not enough funds to increase fees") {
+    withFixture(Seq(10.2 millibtc), ChannelTypes.AnchorOutputsZeroFeeHtlcTx) { f =>
+      import f._
+
+      val (commitTx, anchorTx) = closeChannelWithoutHtlcs(f, aliceBlockHeight() + 30)
+      wallet.publishTransaction(commitTx.tx).pipeTo(probe.ref)
+      probe.expectMsg(commitTx.tx.txid)
+
+      // The feerate is higher for higher block targets
+      val targetFeerate = FeeratePerKw(25_000 sat)
+      setFeerate(FeeratePerKw(3000 sat))
+      setFeerate(targetFeerate, blockTarget = 6)
+      publisher ! Publish(probe.ref, anchorTx)
+      // wait for the commit tx and anchor tx to be published
+      val mempoolTxs1 = getMempoolTxs(2)
+      assert(mempoolTxs1.map(_.txid).contains(commitTx.tx.txid))
+
+      // A new block is found, and the feerate has increased for our block target, but we don't have enough funds to bump the fees.
+      system.eventStream.subscribe(probe.ref, classOf[NotifyNodeOperator])
+      system.eventStream.publish(CurrentBlockCount(aliceBlockHeight().toLong + 15))
+      probe.expectMsgType[NotifyNodeOperator]
+      val mempoolTxs2 = getMempool()
+      assert(mempoolTxs1.map(_.txid).toSet === mempoolTxs2.map(_.txid).toSet)
+    }
+  }
+
+  test("commit tx not confirming, cannot use new unconfirmed inputs to increase fees") {
+    withFixture(Seq(10.2 millibtc), ChannelTypes.AnchorOutputsZeroFeeHtlcTx) { f =>
+      import f._
+
+      val (commitTx, anchorTx) = closeChannelWithoutHtlcs(f, aliceBlockHeight() + 30)
+      wallet.publishTransaction(commitTx.tx).pipeTo(probe.ref)
+      probe.expectMsg(commitTx.tx.txid)
+
+      // The feerate is higher for higher block targets
+      val targetFeerate = FeeratePerKw(25_000 sat)
+      setFeerate(FeeratePerKw(3000 sat))
+      setFeerate(targetFeerate, blockTarget = 6)
+      publisher ! Publish(probe.ref, anchorTx)
+      // wait for the commit tx and anchor tx to be published
+      val mempoolTxs1 = getMempoolTxs(2)
+      assert(mempoolTxs1.map(_.txid).contains(commitTx.tx.txid))
+
+      // Our wallet receives new unconfirmed utxos: unfortunately, BIP 125 rule #2 doesn't let us use that input...
+      wallet.getReceiveAddress().pipeTo(probe.ref)
+      val walletAddress = probe.expectMsgType[String]
+      val walletTx = sendToAddress(walletAddress, 5 millibtc, probe)
+
+      // A new block is found, and the feerate has increased for our block target, but we can't use our unconfirmed input.
+      system.eventStream.subscribe(probe.ref, classOf[NotifyNodeOperator])
+      system.eventStream.publish(CurrentBlockCount(aliceBlockHeight().toLong + 15))
+      probe.expectMsgType[NotifyNodeOperator]
+      val mempoolTxs2 = getMempool()
+      assert(mempoolTxs1.map(_.txid).toSet + walletTx.txid === mempoolTxs2.map(_.txid).toSet)
+    }
+  }
+
+  test("commit tx not confirming, updating confirmation target") {
```

Comment:
There is no log saying "publishing local-anchor with confirmation target in xx blocks" for the initial publication attempt, like there is when bumping: "local-anchor confirmation target is in 15 blocks: bumping fees". Wouldn't that be convenient for debug?

Reply:
Done in https://github.com/ACINQ/eclair/pull/2113/commits/67e0d962f11ca024f2b1b84e9eecd7b74753b4b6